### PR TITLE
Trust Shikaree Z Gambits and Related

### PR DESCRIPTION
**Pull request by neuromancerxi**
* Implements trust spell script for Shikaree Z (commented with TODOs)
* Populate Shikaree Z entries for mob_skill_lists
* Adds Erase to Shikaree Z mob_spell_lists
* Adds mob_skills/Raiden Thrust with SC properties
* Updates mob_skills/Red Lotus Blade to have SC properties
* Adds mobskills/raiden_thrust script

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
Adds scripts and plumbing to support Shikaree Z Trust

## Steps to test these changes
Build an appropriate Test Character
Summon Shikaree Z
Fight Mobs, observe behavior matches Gambits

### CODE REVIEW

File: scripts/globals/spells/trust/shikaree_z.lua [hunk 1]
```diff
@@ -15,6 +23,38 @@ end
 
 spell_object.onMobSpawn = function(mob)
     xi.trust.message(mob, xi.trust.message_offset.SPAWN)
+
+    -- TODO: Load/Apply MODs from mob_pool_mods instead
+    mob:addMod(xi.mod.MPP, 100)
+    mob:addMod(xi.mod.STORETP, 174)
+    mob:addMod(xi.mod.JUMP_TP_BONUS, 164)
```
Comment (zach2good) [inline scripts/globals/spells/trust/shikaree_z.lua:30]: Where have these numbers come from?

File: scripts/globals/spells/trust/shikaree_z.lua [hunk 2]
```diff
@@ -15,6 +23,38 @@ end
...
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.HPP_LT, 50, ai.r.MA, ai.s.HIGHEST, xi.magic.spellFamily.CURE)
+
+    -- TODO: Uses Ancient Circle if the enemy is a dragon
+
+    mob:addSimpleGambit(ai.t.TARGET, ai.c.ALWAYS, 0, ai.r.JA, ai.s.SPECIFIC, xi.ja.JUMP)
+    mob:addSimpleGambit(ai.t.TARGET, ai.c.ALWAYS, 0, ai.r.JA, ai.s.SPECIFIC, xi.ja.HIGH_JUMP)
+    mob:addSimpleGambit(ai.t.SELF, ai.c.HAS_TOP_ENMITY, 0, ai.r.JA, ai.s.SPECIFIC, xi.ja.SUPER_JUMP)
+
+    -- TODO: Hold TP to 2000 before closing SC
+    mob:setTrustTPSkillSettings(ai.tp.CLOSER, ai.s.HIGHEST)
```
Comment (zach2good) [inline scripts/globals/spells/trust/shikaree_z.lua:57]: I think this condition can be left as todo, it's a little difficult

File: scripts/globals/spells/trust/shikaree_z.lua [hunk 3]
```diff
@@ -15,6 +23,38 @@ end
...
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.PARALYSIS, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.PARALYNA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.BLINDNESS, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.BLINDNA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.SILENCE, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.SILENA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.PETRIFICATION, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.STONA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.DISEASE, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.VIRUNA)
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.STATUS, xi.effect.CURSE_I, ai.r.MA, ai.s.SPECIFIC, xi.magic.spell.CURSNA)
+
+    mob:addSimpleGambit(ai.t.PARTY, ai.c.HPP_LT, 50, ai.r.MA, ai.s.HIGHEST, xi.magic.spellFamily.CURE)
+
+    -- TODO: Uses Ancient Circle if the enemy is a dragon
```
Comment (zach2good) [inline scripts/globals/spells/trust/shikaree_z.lua:50]: I think mob:addSimpleGambit(ai.t.TARGET,  ai.c.FAMILY (or SYSTEM) would be pretty easy to add

**Review by zach2good — COMMENTED** [review]
Looking good

File: scripts/globals/spells/trust/shikaree_z.lua (hunk 1 above)
Comment (neuromancerxi) [reply scripts/globals/spells/trust/shikaree_z.lua:30]: Based on BG-Wiki comments, I muddled with TP Returns during testing to aim for the following values:

Melee Swing: ~205 TP Returned
Jump: ~655 TP Returned

File: scripts/globals/spells/trust/shikaree_z.lua (hunk 2 above)
Comment (neuromancerxi) [reply scripts/globals/spells/trust/shikaree_z.lua:57]: To clarify, is the suggestion to remove line 57 until there is an `ai.tp.CLOSER_2000_TP` (or similar) handler?

File: scripts/globals/spells/trust/shikaree_z.lua (hunk 3 above)
Comment (neuromancerxi) [reply scripts/globals/spells/trust/shikaree_z.lua:50]: I plan on iterating through these as I continue working on them in stages (similar to the Adelheid PRs I submitted recently). Leaving notes as I work through it.

File: sql/mob_skill_lists.sql [hunk 4]
```diff
@@ -3504,7 +3504,10 @@ INSERT INTO `mob_skill_lists` VALUES ('TRUST_Prishe',1028,3234); -- Nullifying D
 INSERT INTO `mob_skill_lists` VALUES ('TRUST_Prishe',1028,3235); -- Auroral Uppercut
 INSERT INTO `mob_skill_lists` VALUES ('TRUST_Prishe',1028,3236); -- Knuckle Sandwich
 -- INSERT INTO `mob_skill_lists` VALUES ('TRUST_Ulmia',1029,0);
--- INSERT INTO `mob_skill_lists` VALUES ('TRUST_Shikaree_Z',1030,0);
+INSERT INTO `mob_skill_lists` VALUES('TRUST_Shikaree_Z', 1030, 114); -- Raiden Thrust
+INSERT INTO `mob_skill_lists` VALUES('TRUST_Shikaree_Z', 1030, 118); -- Skewer
+INSERT INTO `mob_skill_lists` VALUES('TRUST_Shikaree_Z', 1030, 119); -- Wheeling Thrust
+INSERT INTO `mob_skill_lists` VALUES('TRUST_Shikaree_Z', 1030, 120); -- Impulse Drive
```
Comment (zach2good) [inline sql/mob_skill_lists.sql]: No additional spaces needed in here

**Review by zach2good — APPROVED** [review]

**Comment by N3ckB3ard** [comment]
image

Seem to receive this error in my log when ShikareeZ uses the ability jump
This image was taken from my map log on x64 - Release

I had my server built on x64 - Debug and this was related to my crash point

**Pull request merged**